### ISSUE TITLE
Reading: Neural audio codecs: how to get audio into LLMs

### ISSUE THREAD

**Issue by josephmaa** [ldoshi/rome-wasnt-built-in-a-day#268]
Article: https://kyutai.org/codec-explainer/

Main idea: Many LLMS convert the speech to text, query the llm, then use text to speech to respond

- Neural audio codecs allows tokenization of audio
- Quantization is not differentiable since it is essentially like assigning a class to the vector, gets ignored in the gradient computation since it's non-differentiable "straight-through" estimator
- Reconstructed images may recreate the same image if the class assignment after quantization is the same
- Residuals are the difference between the predicted and the actual value
- Loss calculated as the log difference between spectrograms (apparently a simple loss function)
- Increasing the number of Residual Vector Quantization levels improve the reconstruction loss
- They also bring up an adversarial GAN for computing loss that improves the audio generation 
- Residual Vector Quantization random dropout also helps because at inference time, the input vector doesn't need all levels in order to recreate a spectrogram with decent interpretability

- Follow up post for loss functions with regards to sound: https://www.soundsandwords.io/audio-loss-functions
- Sound wave can be broken up into a discrete waveform 
- Can drop the phase component commonly on reconstruction of a sound
- L1 and L2 errors, mean absolute error and mean squared error respectively
- Categorical cross entropy can be used (mode collapse can happen with continuous value and the model predicts 0 (between -1 and 1)
- Frobenius norm calculating spectral loss
